client.go

```diff
@@ -95,16 +95,37 @@ func newClient(apiKey string, httpClient *http.Client) *Client {
 	}
 }
 
+func validatePathParameters(params []string) error {
+	invalidParams := []string{}
+	for _, param := range params {
+		if len(strings.TrimSpace(param)) == 0 {
+			invalidParams = append(invalidParams, param)
+		}
+	}
+	if len(invalidParams) > 0 {
+		return &Error{
+			Message: "Operation parameters cannot be empty strings",
+			Class:   ErrorClassClient,
+			Type:    ErrorTypeValidation,
```

Comment:
I think the problem with making this a validation error, is these are mostly shown to the customer. I'd rather this be a BadRequest error or something that isn't mean to be caught.

Reply:
ideally this hits the part of their error handling that just logs it and kills the request with a generic error.

Reply:
Agreed, I used Validation just to continue the discussion with a `recurly.Error` being used. I actually debated defining a new `ErrorTypeX` to be used in cases like this that aren't actual API response based errors